url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/tool/SimpleHttpServer.java

```diff
@@ -0,0 +1,109 @@
+package net.hetimatan.tool;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+
+import net.hetimatan.io.file.KyoroFile;
+import net.hetimatan.io.file.KyoroFileForFiles;
+import net.hetimatan.io.filen.CashKyoroFile;
+import net.hetimatan.net.http.HttpHistory;
+import net.hetimatan.net.http.HttpServerFront;
+import net.hetimatan.net.http.HttpServerTemplate;
+import net.hetimatan.util.event.net.io.KyoroSocket;
+import net.hetimatan.util.http.HttpRequest;
+
+
+public class SimpleHttpServer extends HttpServerTemplate {
+
+	private static SimpleHttpServer server = null;
+	public static void main(String[] args) {
+		if(args.length != 1) {
+			System.out.println("java "+SimpleHttpServer.class.getSimpleName());
+			return;
+		}
+		server = new SimpleHttpServer(new File(args[0]));
+		server.setPort(18080);
+		server.startServer(null);
+	}
+
+	private File mRootDir = null;
+	public SimpleHttpServer(File rootDir) {
+		mRootDir = rootDir;
+	}
+
+	@Override
+	public KyoroFile createResponse(HttpServerFront front, KyoroSocket socket, HttpRequest uri) throws IOException {
+		HttpHistory.get().pushMessage(sId+"#createResponse:"+front.sId+","+uri.getLine().getRequestURI().getMethod()+"\n");
+		File targetFile = new File(mRootDir, uri.getLine().getRequestURI().getMethod());
+		if(!targetFile.exists()) {
+			return createErrorResponse(uri);
+		} else if(targetFile.isDirectory()) {
+			return createDirResponse(uri,targetFile);			
+		} else {
+			return createReturnFileResponse(uri, targetFile);
+		}
+	}
+
+	private KyoroFile createErrorResponse(HttpRequest uri) throws IOException {
+		CashKyoroFile header = new CashKyoroFile(1024);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource: *header*. The resource is referenced at lines 50, 51, 52, 53. The resource is returned at line 54. There are other execution paths that don&#39;t close the resource or return it, for example, when CashKyoroFile.addChunk throws an exception. To prevent this resource leak, close *header* along these other paths before you exit this method.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).